src/Converter_TEST.cc

```diff
@@ -618,6 +619,118 @@ TEST(Converter, RemoveDescendantElement)
+  EXPECT_STREQ(childElem->Name(), "elemB");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemC");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemD");
+
+  // Test removing empty elements
+  // Set up a convert file
+  std::stringstream convertStream;
+  convertStream << "<convert name='elemA'>"
+                << "  <convert name='elemB'>"
+                << "    <convert name='elemC'>"
+                << "      <remove_empty element='elemD'/>"
+                << "    </convert>"
+                << "  </convert>"
+                << "</convert>";
+  tinyxml2::XMLDocument convertXmlDoc;
+  convertXmlDoc.Parse(convertStream.str().c_str());
+  sdf::Converter::Convert(&xmlDoc, &convertXmlDoc);
+
+  tinyxml2::XMLElement *convertedElem =  xmlDoc.FirstChildElement();
+  EXPECT_STREQ(convertedElem->Name(), "elemA");
+  convertedElem = convertedElem->FirstChildElement();
+  ASSERT_NE(nullptr, convertedElem);
+  EXPECT_STREQ(convertedElem->Name(), "elemB");
+  EXPECT_NE(nullptr, convertedElem->FirstChildElement("elemC"));
+  convertedElem = convertedElem->FirstChildElement("elemC");
+  ASSERT_NE(nullptr, convertedElem);
+  auto elemD = convertedElem->FirstChildElement("elemD");
+  ASSERT_NE(elemD, nullptr);
+  while (elemD)
+  {
+    std::string elemValue = elemD->GetText();
+    EXPECT_EQ(elemValue, "D");
+    elemD = elemD->NextSiblingElement("elemD");
+  }
+}
+
+////////////////////////////////////////////////////
+/// Ensure that Converter::Remove function is working with descendant_name
+/// Test removing empty elements only
+TEST(Converter, RemoveEmptyDescendantElement)
+{
+  // Set up an xml string for testing
+  std::string xmlString = getRepeatedXmlString();
+
+  // Verify the xml
+  tinyxml2::XMLDocument xmlDoc;
+  xmlDoc.Parse(xmlString.c_str());
+  tinyxml2::XMLElement *childElem =  xmlDoc.FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemA");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemB");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemC");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemD");
+
+  // Test removing empty elements
+  // Set up a convert file
+  std::stringstream convertStream;
+  convertStream << "<convert name='elemA'>"
+                << "  <convert descendant_name='elemC'>"
+                << "    <remove_empty element='elemD'/>"
+                << "  </convert>"
+                << "</convert>";
+  tinyxml2::XMLDocument convertXmlDoc;
+  convertXmlDoc.Parse(convertStream.str().c_str());
+  sdf::Converter::Convert(&xmlDoc, &convertXmlDoc);
+
+  tinyxml2::XMLElement *convertedElem = xmlDoc.FirstChildElement();
```

Comment:
Add assertion for `convertedElem`.

Reply:
https://github.com/ignitionrobotics/sdformat/pull/720/commits/4df59689dd05d4e8ff20029879777cfc143efacf